cpp/tests/utilities/base_fixture.hpp

```diff
@@ -233,17 +226,18 @@ inline auto parse_cudf_test_opts(int argc, char **argv) {
 /**
  * @brief Macro that defines main function for gtest programs that use rmm
  * 
- * Should be included in every test program that uses rmm allocators.The `main`
- * function is a wrapper around the google test generated `main`, mantaining
- * the original functionality. In addition, the custom `main` function parses
- * the command line to customize test behavior, like allocation mode.
- *
+ * Should be included in every test program that uses rmm allocators since
+ * it maintains the lifespane of the rmm default memory resource.
+ * This `main` function is a wrapper around the google test generated `main`,
+ * mantaining the original functionality. In addition, this custom `main`
+ * function parses the command line to customize test behavior, like the
+ * allocation mode used for creating the default memory resource.
+ * 
  */
 #define CUDF_TEST_PROGRAM_MAIN() int main(int argc, char **argv) {  \
   ::testing::InitGoogleTest(&argc, argv);                           \
   auto const cmd_opts = parse_cudf_test_opts(argc, argv);           \
   auto const rmm_mode = cmd_opts["rmm_mode"].as<std::string>();     \
-  auto const rmm_env = ::testing::AddGlobalTestEnvironment(         \
-    new cudf::test::RmmTestEnvironment(rmm_mode));                  \
+  auto resource = cudf::test::create_memory_resource(rmm_mode);     \
```

Comment:
The name of this function should reflect that the rmm default resource is set. Otherwise @trevorsm7 's suggestion to move `set_default_resource` to main also addresses this perfectly.